spec/controllers/password_retrieval_controller_spec.rb

```diff
@@ -54,26 +54,57 @@
     it 'checks when token does not exist' do
       local_token = 'some random strin'
       local_token_sent_as_parameter = 'randome some'
-      @password_retrival = PasswordReset.new
-      @password_retrival.token = Digest::SHA1.hexdigest(local_token)
-      @password_retrival.user_email = 'example@example.edu'
-      @password_retrival.save!
+      @password_retreival = PasswordReset.new
+      @password_retreival.token = Digest::SHA1.hexdigest(local_token)
+      @password_retreival.user_email = 'example@example.edu'
+      @password_retreival.save!
       request_params = { token: local_token_sent_as_parameter }
       get :check_token_validity, params: request_params
       expect(response).to render_template 'password_retrieval/forgotten'
     end
 
     it 'checks when token is not expired' do
       local_token = 'some random string'
-      @password_retrival = PasswordReset.new
-      @password_retrival.token = Digest::SHA1.hexdigest(local_token)
-      @password_retrival.user_email = 'example@example.edu'
-      @password_retrival.save!
+      @password_retreival = PasswordReset.new
+      @password_retreival.token = Digest::SHA1.hexdigest(local_token)
+      @password_retreival.user_email = 'example@example.edu'
+      @password_retreival.save!
       request_params = { token: local_token }
-      Timecop.freeze(@password_retrival.updated_at + 2.hours) do
+      Timecop.freeze(@password_retreival.updated_at + 2.hours) do
         get :check_token_validity, params: request_params
         expect(response).to render_template 'password_retrieval/reset_password'
       end
     end
   end
+
+  describe 'check if password updated' do
+    it 'check if password and repassword do match' do
+      @user = User.new
+      @user.email = 'example@example.edu'
+      @user.fullname = 'John Doe'
+      @user.name = 'classman'
+      @user.save!
+      @password_retreival = PasswordReset.new
+      @password_retreival.user_email = 'example@example.edu'
+      @password_retreival.save!
+      request_params = { reset: { password: 'AAAAAAAAA123!!', repassword: 'AAAAAAAAA123!!', email: 'example@example.edu' } }
+      post :update_password, params: request_params
+      expect(response).to redirect_to '/'
```

Comment:
Add an expect clause to check if the reset tokens for the user were actually deleted.
Adding this would help:
`expect(PasswordReset.where(user_email: 'example@example.edu')).not_to exist`